.github/workflows/daily-sanitizers.yml

```diff
@@ -0,0 +1,92 @@
+name: daily-sanitizers
+
+on:
+  schedule:
+    - cron: '0 4 * * *' # run at 4 AM UTC
+
+jobs:
+  build:
+    runs-on: [self-hosted, linux, ARM64]
```

Comment:
why do you run it only on the self-hosted machine?

Reply:
matrix:
        # Test of these containers
        container: ["ubuntu-dev:20"]
        build-type: [Debug
        runner: [ubuntu-latest, [self-hosted, linux, ARM64]]
    runs-on: ${{ matrix.runner }}

Reply:
Because the gh runners run out of memory...

Reply:
I am aware but really good mention. The reason I did not pursue this is because of the `costs` of these runners. Normal runners for open source projects get some free time (I think) and then you pay. Whereas with this, you pay every time. As we already pay for the AWS runner I don't see a reason to introduce an extra billing cycle nor can I estimate if it's actually cheaper to run it on gh larger runners :shrug:

Reply:
Ok. we are getting OOM when we are building right? can you use ninja -j4 and see if it helps?

Reply:
I can try but is it really worh it? We still get an extra failure

Reply:
what do you mean we get extra failure?

Reply:
oh you mean to use ubuntu 22 with the normal runners? Sure I can try :)

Reply:
https://github.com/dragonflydb/dragonfly/actions/runs/8098506590/job/22132072377#step:7:973 

Failed no space left again